### PR TITLE
Add EVM to Legacy state or storage(AMDB) logic

### PR DESCRIPTION
**Pull request by JimmyShi22**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -57,6 +57,8 @@ add_subdirectory(evmc)
 add_subdirectory(libinterpreter)
 add_subdirectory(libevm)
 add_subdirectory(libp2p)
+add_subdirectory(liblegacystate)
```
Comment (wheatli) [inline CMakeLists.txt]: In my opinion, legacy means to be dropped while this is not our plan...so you'd better give a more suitable name, such as mptstate

File: libexecutivecontext/LegacyState.h [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    LegacyState(LegacyState const& _s) : m_state(_s.m_state) {}
+
+    /// Copy state object.
+    LegacyState& operator=(LegacyState const& _s)
+    {
+        m_state = _s.m_state;
+        return *this;
+    }
+
+    static OverlayDB openDB(boost::filesystem::path const& _path, h256 const& _genesisHash,
```
Comment (wheatli) [inline libexecutivecontext/LegacyState.h]: The static method makes the multipul ledger harder to  implementaion

File: liblegacystate/Defaults.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/*
+    This file is part of cpp-ethereum.
+
+    cpp-ethereum is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
```
Comment (wheatli) [inline liblegacystate/Defaults.cpp]: I don't think we need this file, specially in multipul ledgers version

File: CMakeLists.txt (hunk 1 above)
Comment (fisco-dev) [reply CMakeLists.txt]: Yes. It's better to name as MptState

Comment (fisco-dev) [reply CMakeLists.txt]: Yes. It's better to name as MptState

File: libexecutivecontext/LegacyState.h (hunk 2 above)
Comment (wheatli) [reply libexecutivecontext/LegacyState.h]: I have found that I misunderstanding the function, it is a tool to open a database and return a DB handler to caller, so it's all right

**Pull request merged**